c2corg_ui/static/js/documentediting.js

```diff
@@ -595,6 +594,45 @@ app.DocumentEditingController.prototype.updateMaxSteps = function(waypointType)
 
 
 /**
+ * Check what properties are missing and tell the user.
+ * @param {Object} doc
+ * @export
+ */
+app.DocumentEditingController.prototype.checkMissingProps = function(doc) {
+  var type = doc.type ? app.utils.getDoctype(doc.type) : this.module_;
+  var fields = app.constants.REQUIRED_FIELDS[type];
+  // understandable structure by alertService
+  var missing = {'data': {'errors': []}};
+  var pass;
+  var field;
+
+  for (var i = 0; i < fields.length; i++) {
+    field = fields[i];
+    pass = false;
+
+    if (field === 'title' || field === 'lang') {
+      pass = (!doc['locales'] || !doc['locales'][0][field]);
+    } else if (field === 'activities') {
+      pass = (!doc['activities'] || doc['activities'].length === 0);
+    } else if (field === 'routes') {
+      pass = (!doc['associations'] || doc['associations']['routes'].length === 0);
+    } else if (field === 'latitude' || field === 'longitude') {
+      pass = (!doc['lonlat'] || doc['lonlat'][field] === (null || undefined));
+    } else if (field === 'date_start') {
+      pass = doc['date_start'] === (null || undefined);
```

Comment:
If that's what you want to check, yes. But note the difference between `==` and `===` in JavaScript.